core/drand_control.go

```diff
@@ -45,6 +45,9 @@ func (d *Drand) InitDKG(c context.Context, in *drand.InitDKGPacket) (*drand.Grou
 	}
 	d.log.Info("init_dkg", "begin", "time", d.opts.clock.Now().Unix(), "leader", true)
 
+	// TODO insert it into the leaderRunSetup, pushDKGInfo as well
+	nc, cancel := context.WithCancel(context.Background())
```

Comment:
will this mean the ctrl-c on the command line now cancels the on-going resharing in the daemon?

Reply:
Nope - it would if I took the context `c` given by gRPC which closes when the other end closes. But since this is in a go routine (and no way to preemptively kill go routine afaik) this code will run until correctly finished.
This is the same logic as what's used for the CTRL-C of "non-leaders" node.